src/audio/midi.rs

```diff
@@ -0,0 +1,146 @@
+use crate::Result;
+use anyhow::anyhow;
+use midir::{MidiInput, MidiInputPort};
+use num_traits::Num;
+use std::sync::mpsc::{self, Receiver};
+use std::thread;
+use std::time::Duration;
+use std::{convert::TryFrom, sync::mpsc::Sender};
+use wmidi::{Channel, ControlFunction, ControlValue, MidiMessage};
+
+/// Listens on the provided port, and sends MIDI messages over a channel. Returns the receiver of that channel.
+pub fn listen_for_input(port_name: &str) -> Result<Receiver<MidiMessage<'static>>> {
+    let port = port(port_name)?;
+
+    println!("MIDI input: {}", port_name);
+
+    let (sender, receiver) = mpsc::channel();
```

Comment:
This is a concept I don't think I'd run into before Rust. A channel is a way of safely sending messages around (usually between threads, in Rust). `mpsc` here means "multiple producer, single consumer". There are other kinds of channels useful for different situations. Technically I only have 1 producer here, but that doesn't matter.

Roughly:

```
sender.send(1)?;

// in another thread

receiver.recv()? // returns `Some(1)`.
```

This function starts listening to the MIDI port, and sends MIDI messages over the channel as it receives them.